crates/cext/src/transpiler/target.rs

```diff
@@ -869,6 +872,474 @@ pub unsafe extern "C" fn qk_target_num_instructions(target: *const Target) -> us
+    target_borrowed.contains_key(name)
+}
+
+/// @ingroup QkTarget
+/// Return the number of properties defined for the specified operation in
+/// the ``QkTarget`` instance, a.k.a. the length of the property map. Returns
+/// an invalid number when not found.
+///
+/// @param target A pointer to the ``QkTarget``.
+/// @param index The index in which the gate is stored.
+///
+/// # Example
+/// ```c
+///     QkTarget *target = qk_target_new(5);
+///     QkTargetEntry *target_entry = qk_target_entry_new(QkGate_H);
+///     qk_target_add_instruction(target, target_entry);
+///
+///     size_t num_props = qk_target_num_properties(target, 0);
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``QkTarget`` is not a valid, non-null pointer to a ``QkTarget``.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_num_properties(target: *const Target, index: usize) -> usize {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target_borrowed = unsafe { const_ptr_as_ref(target) };
+    target_borrowed
+        .get_from_index(index)
+        .map(|(_, v)| v.len())
+        .unwrap_or(usize::MAX)
+}
+
+/// @ingroup QkTarget
+/// Checks if the instruction provided contains properties for the specified qargs.
+///
+/// @param target A pointer to the ``QkTarget``.
+/// @param index The index in which the gate is stored.
+/// @param qargs A pointer to the array of ``uint32_t`` qubit indices to add the
+///     check for, can be a null pointer to check for global properties.
+/// @param num_qubits The length of the qargs array.
+///
+///
+/// # Example
+/// ```c
+///     QkTarget *target = qk_target_new(5);
+///
+///     QkTargetEntry *entry = qk_target_entry_new(QkGate_CX);
+///     uint32_t qargs[2] = {0, 1};
+///     qk_target_entry_add_property(entry, qargs, 2, 0.0, 0.1);
+///
+///     qk_target_add_instruction(target, entry);
+///
+///     bool has_0_1 = qk_target_op_has_qargs(target, 0, qargs, 2); // will be true
+///     bool has_0_2 = qk_target_op_has_qargs(target, 0, (uint32_t *){0, 2}, 2); // will be false
+///     bool has_global = qk_target_op_has_qargs(target, 0, NULL, 0); // will be false
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``QkTarget`` is not a valid, non-null pointer to a ``QkTarget``.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_op_has_qargs(
+    target: *const Target,
+    inst_idx: usize,
+    qargs: *const u32,
+    num_qubits: u32,
+) -> bool {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target_borrowed = unsafe { const_ptr_as_ref(target) };
+    // SAFETY: Per the documentation the qubits pointer is an array of num_qubits
+    // elements or a NULL pointer.
+    let parsed = unsafe { parse_qargs(qargs, num_qubits) };
+    target_borrowed
+        .get_from_index(inst_idx)
+        .map(|(_, inst_map)| inst_map.contains_key(&parsed))
+        .unwrap_or(false)
```

Comment:
IMO, it's better to panic if the index isn't valid, rather than to return `false` (that way we don't hide what is likely a programmer error).

Reply:
Yeah that makes sense here since we're checking for qargs specifically.